### PR TITLE
Enable custom ping messages

### LINKED ISSUE #184: Attach message to ping
Currently, the only means of communicating with a student in the queue is to ping them, which is vague, broadcast, which bothers everyone unnecessarily, or to email them, which can be cumbersome. I would like to be able to attach a message when I ping a student.

Right now, I'd primarily use this to remind individual remote students to add online meeting passwords in their help request descriptions. I imagine this feature might also be helpful during in-person office hours, in order to remind students about adding their physical location.

### PR DESCRIPTION
**Pull request by pl98**
Addresses issue #184 

When an instructor clicks the Ping button, a dialog box will appear with the default message "An instructor is looking for you!" The instructor is able to edit this message and send a notification to the student with a custom ping message.

### CODE REVIEW

File: app/channels/queue_channel.rb [hunk 1]
```diff
@@ -50,6 +50,16 @@ def update_instructor_message(data)
 
   end
 
+  def update_ping_message(data)
```
Comment (mterwill) [inline app/channels/queue_channel.rb:53]: Did you have a use case in mind for why the ping message would be shared across instructors?

File: app/assets/javascripts/components/action.js.jsx [hunk 2]
```diff
@@ -1,7 +1,15 @@
 var Action = React.createClass({
   render: function () {
+    if (this.props.data.title === 'Ping') {
```
Comment (mterwill) [inline app/assets/javascripts/components/action.js.jsx:3]: Could the `Ping` class compose the `Action` component and not the other way around?

**Review by mterwill — COMMENTED** [review]
This looks good! Thank you for the contribution 🙏 

A few small comments.

File: app/assets/javascripts/components/action.js.jsx (hunk 2 above)
Comment (pl98) [reply app/assets/javascripts/components/action.js.jsx]: Fixed!

File: app/channels/queue_channel.rb (hunk 1 above)
Comment (pl98) [reply app/channels/queue_channel.rb]: You're right, it doesn't make sense to structure the code this way so I've changed it. Let me know if this works!

File: app/assets/javascripts/components/action.js.jsx [hunk 3]
```diff
@@ -1,7 +1,9 @@
 var Action = React.createClass({
   render: function () {
     return (
-      <a className={this.props.data.className} onClick={this.props.data.action}>{this.props.data.title}</a>
+      <span style={{ paddingRight: '4px' }}>
```
Comment (mterwill) [inline app/assets/javascripts/components/action.js.jsx:4]: Looks like this is adding some extra spacing to the instructor buttons?

**Review by mterwill — COMMENTED** [review]
One more small comment, otherwise this looks perfect!

File: app/assets/javascripts/components/action.js.jsx (hunk 3 above)
Comment (annieli555) [reply app/assets/javascripts/components/action.js.jsx:4]: Oops, sorry, that actually wasn't needed! I had added this part to address some formatting issues in the request, but it seems like this particular addition was not necessary to fix those issues. l'll push my changes again!

**Comment by mterwill** [comment]
I did notice in testing that the default message flashes briefly as the modal is hiding. Could we maybe use the [`onHidden` callback](https://semantic-ui.com/modules/modal.html#/settings)?

> Is called after a modal has finished hiding animation.

File: app/assets/javascripts/components/course_queue.js.jsx [hunk 4]
```diff
@@ -177,9 +177,9 @@ var CourseQueue = React.createClass({
           this.broadcastInstructorMessage(data);
         } else if (data.action === 'bump'
                    && data.requester_id === this.props.currentUserId) {
-           this.notify(data.bump_by.name + ' is looking for you!', true, {
```
Comment (mterwill) [inline app/assets/javascripts/components/course_queue.js.jsx:180]: Think it'd be useful to keep the instructor's name in there? "Matt says: I'm looking for you" 🤷‍♂️

Comment (pl98) [reply app/assets/javascripts/components/course_queue.js.jsx:180]: Fixed!

**Review by mterwill — APPROVED** [review]
Thank you!

**Comment by mterwill** [comment]
Deployed 🚀
